src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/Tensor.cs

```diff
@@ -0,0 +1,355 @@
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Globalization;
+using System.Linq;
+using System.Reflection;
+using System.Reflection.Metadata;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Runtime.Serialization;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.VisualBasic;
+using static System.Runtime.InteropServices.JavaScript.JSType;
+
+#pragma warning disable CS8601 // Possible null reference assignment.
+#pragma warning disable CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider declaring as nullable.
+#pragma warning disable 8500 // address / sizeof of managed types
+
+namespace System.Numerics.Tensors
+{
+    public sealed class Tensor<T>
+        : IDisposable,
+          ITensor<Tensor<T>, T>
+        where T : IEquatable<T>
+    {
+        /// <summary>A byref or a native ptr.</summary>
+        internal readonly T[] _values;
+        /// <summary>The number of elements this Tensor contains.</summary>
+        internal readonly nint _linearLength;
+        /// <summary>The lengths of each dimension.</summary>
+        internal readonly nint[] _lengths;
+        internal readonly nint[] _strides;
+        /// <summary>If the backing memory is permanently pinned (so not just using a fixed statement).</summary>
+        internal readonly bool _isPinned;
+
+        /// <summary>
+        /// Creates a new empty Tensor.
+        /// </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal Tensor()
+        {
+            _linearLength = 0;
+            _values = [];
+            _lengths = [];
+            _strides = [];
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal Tensor(T[] values, nint[] lengths, bool isPinned)
+        {
+            _linearLength = SpanHelpers.CalculateTotalLength(ref lengths);
+
+            Debug.Assert(_linearLength >= 0);
+
+            _values = values;
+            _lengths = lengths;
+            _strides = SpanHelpers.CalculateStrides(Rank, _lengths);
+            _isPinned = isPinned;
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal Tensor(T[] values, nint[] lengths, nint[] strides, bool isPinned)
+        {
+            _linearLength = SpanHelpers.CalculateTotalLength(ref lengths);
+
+            Debug.Assert(_linearLength >= 0);
+
+            _values = values;
+            _lengths = lengths;
+            _strides = strides;
+            if (strides == Array.Empty<nint>())
+                _strides = SpanHelpers.CalculateStrides(Rank, lengths);
+            _isPinned = isPinned;
+
+        }
+
+        // IDisposable
+
+        public void Dispose()
+        {
+        }
```

Comment:
Why does this sealed class implement IDisposable if its Dispose is a nop?

Reply:
Depending on how we end up supporting Tensors with native memory though it may need to come back so I figure having it there from the start would be better. Its removed for now though and we can re-add it if we really need to.